enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+* a list of NTP sources (hostname or IP) to be added to all cluster
+hosts. They are added to any NTP sources that were configured through other
+means.
+* a list of SSH public keys that will be added to all agents for use in debugging.
+* pull secret reference
+* a label selector for Agents. This is how Agent resources are identified as
+belonging to a particular InstallEnv.
+* labels to add to Agent resources upon creation. These labels should match
+the label selector above.
+* Array of ClusterDeployment references. This is a whitelist of ClusterDeployments
+that are authorized to consume Agents from this InstallEnv. See the section below
+on [Namespaces and Multi-tenancy](#namespaces-and-multi-tenancy) for details.
+
+Status
+* a URL to download the discovery ISO produced for this InstallEnv.
+
+**Agent**
+Agent is a new resource, part of the assisted installer's new operator, that
+represents a host that is destined to become part of an OpenShift cluster, and
+is running an agent that is able to run inspection and installation tasks. It
+correlates to the "host" resource in the Assisted Installer's REST API.
+
+The details of this API are available [as
+code](https://github.com/openshift/assisted-service/pull/861), with some
+details yet to be changed.
+
+Summary of fields:
+
+Spec
+* Role that the Node will have
+* Hostname
+* Approved
+* Install device, set automatically when there is platform-integrated
+automation booting the discovery ISO, or otherwise set by the user in a
+boot-it-yourself scenario.
+
+Status
+* Comprehensive inspection and validation data
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include a new `AgentBareMetal` platform that
+contains such fields as:
+
+* API VIP
+* API VIP DNS name
+* IngressVIP
+
+A new `InstallStrategy` section of the Spec enables the API to describe a way
+of installing a cluster other than the default of using `openshift-install`.
+The new field has an "agent" option that would be common to any Agent-driven
+installation, regardless of platform. Fields include:
+
+* AgentSelector, a label selector for identifying which Agent resources should
+be included in the cluster.
+* ProvisionRequirements, where the API user can specify how many agents to
+expect before beginning installation for each of the control-plane and worker
+roles.
+
+The details of this were discussed on a Hive
+[pull request](https://github.com/openshift/hive/pull/1247).
+
+#### REST API Access
+
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download discovery ISO and rootfs: These files must be available for download
+via HTTP, either directly by users or by a BMC.  Because BMCs do not pass
```

Comment:
HTTPS? i think that http is fine for testing, but for production we may aim to use https...

Reply:
there are BMCs that do not support http download. This is the case for several models of SuperMicro, who work using SMB Virtual Media. We need some step here, to download from AI endpoint, and place to the right SMB share.

Reply:
I agree about https. I don't think "http" here is meant to preclude https. It's http either way, just with or without TLS. We should definitely make sure this works with TLS.

Thanks for raising the SMB question. I'll check on whether we need to support that right now. That may otherwise need to fall into the "boot it yourself" category.